File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        int tc=0;
        int l = people.size();
        sort(people.begin(),people.end());
        int i=0,j=people.size()-1;
        //cout << i << j << "ij" << endl;
        while(i<=j){
            if(people[i]+people[j]<=limit){
                tc++;
                // cout << people[i] << " " << people[j] << endl;
                // cout << tc << endl;
                i++;
                j--;
            }
            else{
               tc++;
               j--;
               // cout << people[i] << " " << people[j] << "j" << endl;
               // cout << tc << endl;
            }
        }
       return tc;    
    }
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        int min_boats = 0;
        Arrays.sort(people);
        int i = 0;
        int j = people.length-1;
        System.out.println(people.length);
        while(i<j){
            // System.out.println(i + " " + j + " " + people[i] + " " + people[j]);
            if(people[i] == limit || people[j] == limit){
                if(people[i]==limit){
                    i++;
                    min_boats++;
                }
                if(people[j]==limit){
                    j--;
                    min_boats++;
                }
            }
             else if(people[i] + people[j] <= limit){
                min_boats++;
                i++;
                j--;
            }
            else if(people[i] + people[j] > limit){
                j--;
                min_boats++;
            }
            
        }
        if(i==j){
           min_boats++; 
        }
        return min_boats;
    }
}